src/Stack/Types/PackageDump.hs

```diff
@@ -31,4 +31,6 @@ data InstalledCacheEntry = InstalledCacheEntry
 instance Store InstalledCacheEntry
 
 installedCacheVC :: VersionConfig InstalledCacheInner
-installedCacheVC = storeVersionConfig "installed-v1" "GGyaE6qY9FOqeWtozuadKqS7_QM="
+-- FIXME would be really nice to avoid changing this hash...
+--installedCacheVC = storeVersionConfig "installed-v1" "GGyaE6qY9FOqeWtozuadKqS7_QM="
```

Comment:
The version tag should be bumped if the serialization has changed.  I haven't looked over this change very thoroughly, but can we be sure that the serialization is the same as before?  If so, then you can just change the hash and keep the version tag the same, and things should just work.

What's the motivation for switching over to Cabal types?  To me it'd make sense to minimize dependencies on Cabal api.